crates/provider/src/builder.rs

```diff
@@ -236,20 +313,80 @@ impl<L, N> ProviderBuilder<L, N> {
 
     /// Build this provider with an Hyper HTTP transport.
     #[cfg(feature = "hyper")]
-    pub fn on_hyper_http(self, url: url::Url) -> Result<L::Provider, TransportError>
+    pub fn on_hyper_http(self, url: url::Url) -> Result<F::Provider, TransportError>
     where
         L: ProviderLayer<
             crate::HyperProvider<N>,
             alloy_transport_http::Http<alloy_transport_http::HyperClient>,
             N,
         >,
+        F: TxFiller<N>
+            + ProviderLayer<
+                L::Provider,
+                alloy_transport_http::Http<alloy_transport_http::HyperClient>,
+                N,
+            >,
         N: Network,
     {
         let client = ClientBuilder::default().hyper_http(url);
         Ok(self.on_client(client))
     }
 }
 
+// Enabled when the `anvil` feature is enabled, or when both in test and the
+// `reqwest` feature is enabled.
+#[cfg(any(all(test, feature = "reqwest"), feature = "anvil"))]
+impl<L, F> ProviderBuilder<L, F, Ethereum> {
+    /// Build this provider with anvil, using an Reqwest HTTP transport.
+    pub fn on_anvil(self) -> (F::Provider, alloy_node_bindings::AnvilInstance)
+    where
+        F: TxFiller<Ethereum>
+            + ProviderLayer<L::Provider, alloy_transport_http::Http<reqwest::Client>, Ethereum>,
+        L: ProviderLayer<
+            crate::ReqwestProvider<Ethereum>,
+            alloy_transport_http::Http<reqwest::Client>,
+            Ethereum,
+        >,
+    {
+        let anvil = alloy_node_bindings::Anvil::new().spawn();
+        let url = anvil.endpoint().parse().unwrap();
+
+        (self.on_http(url).unwrap(), anvil)
+    }
+
+    /// Build this provider with anvil, using an Reqwest HTTP transport. This
+    /// function configures a signer backed by anvil keys, and is intended for
+    /// use in tests.
+    #[allow(clippy::type_complexity)]
+    pub fn on_anvil_with_signer(
+        self,
+    ) -> (
+        <JoinFill<F, SignerFiller<alloy_network::EthereumSigner>> as ProviderLayer<
+            L::Provider,
+            alloy_transport_http::Http<reqwest::Client>,
+        >>::Provider,
+        alloy_node_bindings::AnvilInstance,
+    )
+    where
+        L: ProviderLayer<
+            crate::ReqwestProvider<Ethereum>,
+            alloy_transport_http::Http<reqwest::Client>,
+            Ethereum,
+        >,
+        F: TxFiller<Ethereum>
+            + ProviderLayer<L::Provider, alloy_transport_http::Http<reqwest::Client>, Ethereum>,
```

Comment:
hehe, fine with this since all of this is internal complexity

Reply:
just don't look at this impl block lol